tools/mcp/dotnet/AzureSDKDevToolsMCP/Services/DevOpsService.cs

```diff
@@ -0,0 +1,234 @@
+
+        public DevOpsService()
+        {
+            // Connect to Azure DevOps using managed identity
+            var token = (new DefaultAzureCredential()).GetToken(new TokenRequestContext(["499b84ac-1321-427f-aa17-267ca6975798/.default"]));
+            var connection = new VssConnection(new Uri(devOpsUrl), new VssOAuthAccessTokenCredential(token.Token));
+            buildClient = connection.GetClient<BuildHttpClient>();
+            workItemClient = connection.GetClient<WorkItemTrackingHttpClient>();
+        }
+
+        public async Task<ReleasePlan> GetReleasePlan(int workItemId)
+        {
+            Console.WriteLine($"Fetching release plan work with id {workItemId}");
+            var workItem = await workItemClient.GetWorkItemAsync(workItemId);
+            if (workItem?.Id == null)
+                throw new InvalidOperationException($"Work item {workItemId} not found.");
+            var releasePlan = MapWorkItemToReleasePlan(workItem);
+            releasePlan.WorkItemUrl = workItem.Url;
+            releasePlan.WorkItemId = workItem?.Id ?? 0;
+            return releasePlan;
+        }
+
+        private static ReleasePlan MapWorkItemToReleasePlan(WorkItem workItem)
+        {
+            Console.WriteLine($"Work item details: {workItem.Fields.ToString()}");
+            var releasePlan = new ReleasePlan()
+            {
+                WorkItemId = workItem.Id ?? 0,
+                WorkItemUrl = workItem.Url,
+                Title = workItem.Fields.TryGetValue("System.Title", out object? value) ? value?.ToString() ?? string.Empty : string.Empty,
+                Status = workItem.Fields.TryGetValue("System.State", out value) ? value?.ToString() ?? string.Empty : string.Empty,
+                ServiceTreeId = workItem.Fields.TryGetValue("Custom.ServiceTreeID", out value) ? value?.ToString() ?? string.Empty : string.Empty,
+                ProductTreeId = workItem.Fields.TryGetValue("Custom.ProductServiceTreeID", out value) ? value?.ToString() ?? string.Empty : string.Empty,
+                SDKReleaseMonth = workItem.Fields.TryGetValue("Custom.SDKReleaseMonth", out value) ? value?.ToString() ?? string.Empty : string.Empty,
+                IsManagementPlane = workItem.Fields.TryGetValue("Custom.MgmtScope", out value) ? value?.ToString() == "Yes" : false,
+                IsDataPlane = workItem.Fields.TryGetValue("Custom.DataScope", out value) ? value?.ToString() == "Yes" : false
+            };
+            return releasePlan;
+        }
+
+        public async Task<List<ReleasePlan>> GetReleasePlans(string serviceTreeId, string productTreeId, string pullRequest)
+        {
+            var query = $"SELECT [System.Id] FROM WorkItems WHERE [System.TeamProject] = '{releaseProject}' AND [System.WorkItemType] = 'Release Plan' AND [Custom.ServiceTreeID] = '{serviceTreeId}' AND [Custom.ProductServiceTreeID] = '{productTreeId}' AND [System.State] NOT IN ('Abandoned', 'Duplicate', 'Finished')";
```

Comment:
Can we cast `productTreeId` as a guid (or whatever the format needs to be)? Or does wiql protect against injection? Right now it's just running queries with the user's auth context, but it's probably worth getting into the habit of really guarding/validating our inputs coming from AI+users.